### PR TITLE
major: various fixes, codebase improvements/cleanup, packages/sdk update

### PR DESCRIPTION
**Pull request by bryanpth**
#### Updates

- `.NET SDK` from `7.0` to `8.0`
- `C# language` from `9.0` to `12.0`
- and all packages to their latest available versions. (e.g: discord.net, etc..)

#### Fixes

- Fixed `/compare` not properly returning when the same player was asked to be compared.
- Fixed a typo in the alliance rename command output.
- Fixed non-awaited warning for compensation tasks.
- Fixed pagniation not working in DMs.

#### Improvements

- Improved many publicly available commands response ouput.
- Improved almost every moderation commands and logging response ouput.
- Improved various commands response strings.
- Added usernames for authorized discord user ids, this allows to easily identify which user has empowered permissions.
- Added missing `readonly` modifiers on various files.
- Made slight improvements to code readability on various files.
- Made others various miscellaneous changes that improves codebase quality.

#### Documentation

- Updates `README.md` for a clearer output.

#### Codebase cleanup

- Deleted unused Dashboard project.
- Added `.editorconfig` to ensure proper code quality.
- Added `.gitattributes` to ensure normalized line endings to lf internally.

#### Continuous integration

- Deleted previous `console.yml` and `dashboard.yml` workflows.
- Added `ci.yaml`, which ensure proper code formatting and warning/error free application compiling.

--

- [x] Tested all publicly available commands.

### CODE REVIEW

File: src/AdvancedBot.Core/Commands/Preconditions/RequirePrivateList.cs [hunk 1]
```diff
@@ -8,7 +8,15 @@ namespace AdvancedBot.Core.Commands.Preconditions
 {
     public class RequirePrivateList : PreconditionAttribute
     {
-        private List<ulong> _userIds = new List<ulong>() { 202095042372829184, 942849642931032164, 209801906237865984, 180676108088246272, 356060824223350784, 275698828974489612 };
+        private readonly List<ulong> _userIds = new() 
+        { 
+            202095042372829184, // svr333
+            942849642931032164, // lifecoder
+            209801906237865984, // kibble00
```
Comment (svr333) [inline src/AdvancedBot.Core/Commands/Preconditions/RequirePrivateList.cs]:
```suggestion
```

File: src/AdvancedBot.Core/Commands/Preconditions/RequireStagingList.cs [hunk 2]
```diff
@@ -8,7 +8,16 @@ namespace AdvancedBot.Core.Commands.Preconditions
 {
     public class RequireStagingList : PreconditionAttribute
     {
-        private List<ulong> _userIds = new List<ulong>() { 202095042372829184, 942849642931032164, 209801906237865984, 180676108088246272, 356060824223350784, 275698828974489612, 240402743443980288 };
+        private readonly List<ulong> _userIds = new() 
+        { 
+            202095042372829184, // svr333
+            942849642931032164, // lifecoder
+            209801906237865984, // kibble00
```
Comment (svr333) [inline src/AdvancedBot.Core/Commands/Preconditions/RequireStagingList.cs]:
```suggestion
```

File: src/AdvancedBot.Core/Commands/Preconditions/RequirePrivateList.cs (hunk 1 above)
Comment (svr333) [reply src/AdvancedBot.Core/Commands/Preconditions/RequirePrivateList.cs]: Feel free to add yourself though

File: src/AdvancedBot.Core/Commands/TopModule.cs [hunk 3]
```diff
@@ -19,23 +19,27 @@ public class TopModule : InteractionModuleBase<SocketInteractionContext>
         public PaginatorService Paginator { get; set; }
         public LogService LogService { get; set; }
 
-        public readonly List<ulong> PowerUsers = new List<ulong>() { 202095042372829184, 209801906237865984, 942849642931032164, 180676108088246272, 356060824223350784, 275698828974489612 };
-        public readonly List<ulong> SemiPowerUsers = new List<ulong>() { };
+        public readonly List<ulong> PowerUsers = new() 
+        { 
+            202095042372829184, // svr333
+            942849642931032164, // lifecoder
+            209801906237865984, // kibble00
```
Comment (svr333) [inline src/AdvancedBot.Core/Commands/TopModule.cs]:
```suggestion
```

File: src/AdvancedBot.Core/Services/LogService.cs [hunk 4]
```diff
@@ -65,115 +65,116 @@ public Embed GetEmbedForLog(Log log, User victim)
...
+
+                LogAction.UpdateEmail or LogAction.UpdateName or LogAction.GetFull or LogAction.GetChipsBought or
+                LogAction.MakeUserAllianceOwner or LogAction.GetWarlogs
+                    => new Color(15710778),
+
+                LogAction.AddChips or LogAction.AddItem or LogAction.RenameAlliance
+                    => Color.Blue,
+
+                _ => Color.Blue
+            };
```
Comment (svr333) [inline src/AdvancedBot.Core/Services/LogService.cs]: Damn this is new syntax for me, looks cool

File: src/AdvancedBot.Core/Commands/Preconditions/RequirePrivateList.cs (hunk 1 above)
Comment (bryanpth) [reply src/AdvancedBot.Core/Commands/Preconditions/RequirePrivateList.cs]: > Feel free to add yourself though

thanks, will do!

File: src/AdvancedBot.Core/Commands/Modules/ModerationModule.cs [hunk 5]
```diff
@@ -442,110 +485,156 @@ public async Task EnableMaintenanceAsync(uint minutes = 60)
         [SlashCommand("compensatechips", "Compensate chips to all users")]
         public async Task CompensateChips(uint amount)
         {
-            GLClient.Production.CompensateChips(amount);
+            _ = GLClient.Production.CompensateChips(amount);
```
Comment (svr333) [inline src/AdvancedBot.Core/Commands/Modules/ModerationModule.cs]:
```suggestion
            await GLClient.Production.CompensateChips(amount);
```

File: src/AdvancedBot.Core/Commands/Modules/ModerationModule.cs [hunk 6]
```diff
@@ -442,110 +485,156 @@ public async Task EnableMaintenanceAsync(uint minutes = 60)
...
 
-            await ModifyOriginalResponseAsync(x => x.Content = $"Send out compensation of {amount} chips to everyone!");
+            await ModifyOriginalResponseAsync(msg => msg.Content = $"Sending out compensation of {amount} chips to everyone!");
         }
 
         [SlashCommand("compensateitems", "Compensate items to all users")]
         public async Task CompensateItems(string sku, uint amount)
         {
-            GLClient.Production.CompensateItems(sku, amount);
+            _ = GLClient.Production.CompensateItems(sku, amount);
```
Comment (svr333) [inline src/AdvancedBot.Core/Commands/Modules/ModerationModule.cs]:
```suggestion
            await GLClient.Production.CompensateItems(sku, amount);
```

**Review by svr333 — COMMENTED** [review]
Im pretty sure I just forgot to await these, rest of new stuff looks good

**Comment by bryanpth** [comment]
> Im pretty sure I just forgot to await these, rest of new stuff looks good

oh, i assumed compensation task would take a while so you weren't awaiting them to not block the rest of the execution, thanks for checking it though

**Comment by svr333** [comment]
Oh actually now that you mention it that is probably what I did, whoops, will revert those commits

**Comment by bryanpth** [comment]
> Oh actually now that you mention it that is probably what I did, whoops, will revert those commits

alright, reverted those for ya dw

**Comment by bryanpth** [comment]
`AdvancedBot.Dashboard` doesn't seem to be used as it seems to be startup template right now.
Do you mind if I was to delete it to fruther clean up the codebase?

**Comment by svr333** [comment]
Originally, 3 years ago the dashboard was being worked on by twobits to have an admin panel for GLR on it.
But yeah I dont think we actually need one for now so you can delete it

**Review by svr333 — APPROVED** [review]
Everything looks good to me, thanks for all the refactoring!

**Pull request merged**